### PR TITLE
Composition: Fix remote stories by passing refs to virtual module safely

### PR DESCRIPTION
**Pull request by evanroane**
Issue: https://github.com/storybookjs/storybook/issues/10509

## What I did
In `manager-webpack.config.js` I changed ln. 58 to make sure that the single quote characters (`'`) are replaced in the template file and on ln. 59 I added a call to `JSON.stringify` so that the `refs` object is safely converted to a string before `String.prototype.replace` is called since calling it with an object returns the string `"[object Object]"`.

## How to test

- Is this testable with Jest or Chromatic screenshots? No.
- Does this need a new example in the kitchen sink apps? No.
- Does this need an update to the documentation? No.

To test follow the steps in https://github.com/storybookjs/storybook/issues/10509 on the `next` branch and with the code in this pull request. You will notice that the `undefined` value is no longer in the URLs of remote storybooks.

### CODE REVIEW

File: lib/core/src/server/manager/manager-webpack.config.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export default ({
       refs
         ? new VirtualModulePlugin({
             [path.resolve(path.join(configDir, `generated-refs.js`))]: refsTemplate.replace(
-              `'{{refs}}'`,
+              /'{{refs}}'/g,
```
Comment (ndelangen) [inline lib/core/src/server/manager/manager-webpack.config.js:58]: I'm unsure why this is necessary, `'{{refs}}'` only occurs once in the template?

**Comment by ndelangen** [comment]
I think we both found the same bug, and fixed it, Your PR seems to also change the search-term to a regex, did you have a reason for that? I'd like to know.

**Comment by evanroane** [comment]
I went for a regex here because I wanted to capture the single quote characters (`'`) that are in the template. I typically don't use a template literal to form a string that doesn't need multi-line support or expression interpolation. 

While the template literal seems to work just fine, I prefer to use the regex since it seems the more apt object for the job.

File: lib/core/src/server/manager/manager-webpack.config.js (hunk 1 above)
Comment (evanroane) [reply lib/core/src/server/manager/manager-webpack.config.js:58]: true, the `g` is not needed

**Comment by evanroane** [comment]
Feel free to merge this if you think a regex object is a better option, otherwise I'm just glad the bug is fixed 👍

**Comment by evanroane** [comment]
I had actually thought to set refs in the template file to something nullish but still valid JS
```
addons.setConfig({
  refs: undefined,
});
```

and then in the `replace` call in the manager webpack can use the string `'undefined'` a lot of the quotation wrangling disappears that way

**Comment by ndelangen** [comment]
That feels a lot more risky to me tbh, the `'{{refs}}'` makes it pretty clear to me there's something getting replaced, and you can search the codebase for where that might be happening.

**Comment by ndelangen** [comment]
Thank you for the effort, time and help regardless @evanroane 

It's appreciated!

**Pull request closed without merging**